### PR TITLE
Don't close message channel if client doesn't want to recive any longer

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
If the plugin gets disabled while a reviving channel created by the broker is waiting for a message, a double `close` happens as the broker closes the channel and then the consumer closes it again in `UnsubscribeX`. Given that only the sender should close a channel, `UnsubscribeX` doesn't do that with this PR.

The PR also includes various other fixes.

#### Ticket Link
None

### CODE REVIEW

File: server/plugin/flows.go [hunk 1]
```diff
@@ -694,13 +694,16 @@ func (fm *FlowManager) submitWebhook(f *flow.Flow, submitted map[string]interfac
...
-	case event := <-ch:
-		if *event.HookID == *hook.ID {
-			break
+	var found bool
+	for !found {
+		select {
+		case event, ok := <-ch:
+			if ok && event != nil && *event.HookID == *hook.ID {
+				found = true
+			}
```
Comment (hanzei) [inline server/plugin/flows.go:703]: The code would exit if an event for a different user was received.

**Review by catalintomai — APPROVED** [review]

File: server/plugin/flows.go [hunk 2]
```diff
@@ -694,13 +694,16 @@ func (fm *FlowManager) submitWebhook(f *flow.Flow, submitted map[string]interfac
...
-	select {
-	case event := <-ch:
-		if *event.HookID == *hook.ID {
-			break
+	var found bool
+	for !found {
+		select {
+		case event, ok := <-ch:
+			if ok && event != nil && *event.HookID == *hook.ID {
+				found = true
```
Comment (mickmister) [inline server/plugin/flows.go:702]: I know the loop is set to use `found` as its terminating value, but do we want to include the `break` here for readability, or does that cause us to exit the `switch` only?

**Review by mickmister — APPROVED** [review]
LGTM :+1:

File: server/plugin/flows.go (hunk 2 above)
Comment (hanzei) [reply server/plugin/flows.go:702]: `break` only exists the `select` statement. 

> A "break" statement terminates execution of the innermost ["for"](https://go.dev/ref/spec#For_statements), ["switch"](https://go.dev/ref/spec#Switch_statements), or ["select"](https://go.dev/ref/spec#Select_statements) statement within the same function.

https://go.dev/ref/spec#Break_statements

**Comment by hanzei** [comment]
@DHaussermann Are you with merging with PR without further testing as the bug it fixed is minor and hard to reproduce?

**Comment by hanzei** [comment]
Merging without QA testing per https://github.com/mattermost/mattermost-plugin-github/pull/541#issuecomment-1054025620

**Pull request merged**